src/main/java/com/example/demo/dto/BookSearchParametersDto.java

```diff
@@ -0,0 +1,4 @@
+package com.example.demo.dto;
+
+public record BookSearchParametersDto(String titlePart, String authorPart) {
```

Comment:
```suggestion
public record BookSearchParametersDto(String[] title, String[] author) {
```
better pass arrays for multiple searches